### PR TITLE
Comments

### PR DESCRIPTION
**Pull request by RKelley1**
# User can successfully create comments on images/files

## Functionality Description
 - On image load, if no comment object has been created, one is created. Comment list is empty
 - If a comment object exists, comment list is populated with existing comments from comment object.
- if a user creates a comment, the new comment will be displayed on the comment list and also saved into the database.
## Package Changes
- none
## Tests Added
- none

### CODE REVIEW

File: imports/components/commentForm.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+      this.props.saveComment(commentData);
+  }
+
+  render() {
+    return (
+      <form onSubmit={this.handleFormSubmit}>
+        <input type="text" ref="author" placeholder="enter your name (optional)" />
+        <br/>
+        <input type="text" ref="comment" placeholder="comment..." />
```
Comment (kmeyerhofer) [inline imports/components/commentForm.js]: I think a `<textarea>` could work better for a comment box. 

`<textarea ref="comment">Comment...</textarea>` for example.

File: imports/components/fileList.js [hunk 2]
```diff
@@ -24,17 +26,25 @@ class FileList extends Component {
...
+
+            <CommentBox
+              key={generateURL()}
+              id={file._id}
+            />
+          </div>
         );
-      });
+      }
+    ); // end of file map
```
Comment (kmeyerhofer) [inline imports/components/fileList.js]: Clean up comment

File: server/main.js [hunk 3]
```diff
@@ -21,10 +22,38 @@ Meteor.methods({
...
+  saveComments(id, comment) {
+    try {
+      MongoComments.update({_id: id}, {$push: {comments: comment}});
+    } catch(error) {
+      console.log(error);
+    }
+
+  },
+
+  addComments(id, comment) {  // done
```
Comment (kmeyerhofer) [inline server/main.js]: In `addComments` and `saveComments` I think we need to check the user's input so they can't add any malicious code to our database. 

Maybe there's a text cleaner type package that can help us escape any dangerous characters that could be saved to the database.

File: imports/components/commentBox.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    loaded: false,
+  }
+
+  componentDidMount = () => {
+    if ( MongoComments.findOne({_id: this.state.fileID}) === undefined) {
+      this.createCommentObj();
+    }
+  }
+
+  createCommentObj = () => {
```
Comment (kmeyerhofer) [inline imports/components/commentBox.js]: This function, `createCommentObj`, seems unnecessary. Does it create a blank comment object in the database for each file?

File: imports/components/commentBox.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+class CommentBox extends Component {
+  state = {
+    comments: [],
+    fileID: this.props.id,
+    loaded: false,
+  }
+
+  componentDidMount = () => {
+    if ( MongoComments.findOne({_id: this.state.fileID}) === undefined) {
```
Comment (kmeyerhofer) [inline imports/components/commentBox.js]: This check seems unnecessary when we utilize `withTracker` down below. Any updates in the database are noted by `withTracker` and should modify `CommentBox`'s props, rerendering the component. I also think direct queries to the database on the client side aren't allowed, I could be wrong though.

File: server/main.js (hunk 3 above)
Comment (RKelley1) [reply server/main.js]: oh good call i'll look for something

Comment (RKelley1) [reply server/main.js]: https://github.com/hapijs/joi

Comment (RKelley1) [reply server/main.js]: this could be a library we use

File: imports/components/commentList.js [hunk 6]
```diff
@@ -11,10 +12,17 @@ export class CommentList extends Component {
...
-      </div>
-    );
+    if(this.props.comments && this.props.comments.length > 0){
+      return(
+        <div>
+          {this.commentsMap()}
+        </div>
+      );
+    } else {
+      return(
```
Comment (kmeyerhofer) [inline imports/components/commentList.js]: I believe returning `null` is better than an empty `<div>`, would you agree?

File: imports/components/commentList.js [hunk 7]
```diff
@@ -3,6 +3,7 @@ import {Comment} from './comment.js';
 
 export class CommentList extends Component {
   commentsMap = () => {
+    console.log(this.props.comments);
```
Comment (kmeyerhofer) [inline imports/components/commentList.js]: Remove `console.log` please.

File: server/main.js (hunk 3 above)
Comment (kmeyerhofer) [reply server/main.js]: https://www.npmjs.com/package/validator

The methods `escape(input)` is the method I was considering using. It replaces <, >, &, ', " and / with HTML entities, disallowing the browser from rendering elements or the database from executing commands.

I'll make an issue of this and we'll fix it in another branch.

**Pull request merged**